config/rbac/kustomization.yaml

```diff
@@ -16,3 +16,9 @@ resources:
 # - auth_proxy_role.yaml
 # - auth_proxy_role_binding.yaml
 # - auth_proxy_client_clusterrole.yaml
+# For each CRD, "Editor" and "Viewer" roles are scaffolded by
```

Comment:
did you use correct kubebuilder version https://github.com/migtools/oadp-non-admin/blob/ca7e290f36a116593ba8255125baf3b8b561dc6c/docs/architecture.md#kubebuilder ? I did not expect this change

Reply:
The scaffold was using operator-sdk with the following versions:

```shell
# operator-sdk version
operator-sdk version: "v1.37.0", commit: "819984d4c1a51c8ff2ef6c23944554148ace0752", kubernetes version: "1.29.0", go version: "go1.21.13", GOOS: "linux", GOARCH: "amd64"

# kubebuilder version
Version: main.version{KubeBuilderVersion:"4.2.0", KubernetesVendor:"1.31.0", GitCommit:"c7cde5172dc8271267dbf2899e65ef6f9d30f91e", BuildDate:"2024-08-17T09:41:45Z", GoOs:"linux", GoArch:"amd64"}

```

See the commit message:
  https://github.com/migtools/oadp-non-admin/pull/115/commits/6d9db57d78a263be9cc1cc1b07a66d4d93e7b210

Reply:
this project does not use operator-sdk

Reply:
The generated api is same as under the hood operator-sdk calls kubebuilder, question is if I should use older kubebuilder version instead of newer one?

Reply:
I am currently rebasing this on top of kubebuilder generated scaffold with the same version 1.34.0. Maybe it makes sense to use same version, so later we can upgrade to newer one from same base.